lib/charms/traefik_k8s/v1/ingress.py

```diff
@@ -152,23 +175,27 @@ def __init__(self, charm: CharmBase, relation_name: str = DEFAULT_RELATION_NAME)
 
         observe = self.framework.observe
         rel_events = charm.on[relation_name]
-        observe(rel_events.relation_created, self._handle_relation)
-        observe(rel_events.relation_joined, self._handle_relation)
-        observe(rel_events.relation_changed, self._handle_relation)
-        observe(rel_events.relation_broken, self._handle_relation_broken)
+        observe(rel_events.relation_created, self._handle_relation_available)
+        observe(rel_events.relation_joined, self._handle_relation_available)
+        observe(rel_events.relation_changed, self._handle_relation_available)
+        observe(rel_events.relation_broken, self._handle_relation_unavailable)
+        observe(rel_events.relation_departed, self._handle_relation_unavailable)
         observe(charm.on.leader_elected, self._handle_upgrade_or_leader)
         observe(charm.on.upgrade_charm, self._handle_upgrade_or_leader)
```

Comment:
In most other charms, `_on_whatever` seems to be the consensus style of naming handlers. This is purely stylistic however, so feel free to dismiss it. :)